Assets/PatchKit Patcher/Scripts/AppData/Remote/RemoteMetaData.cs

```diff
@@ -15,27 +13,37 @@ public class RemoteMetaData : IRemoteMetaData
         private readonly MainApiConnection _mainApiConnection;
         private readonly KeysApiConnection _keysApiConnection;
 
-        public RemoteMetaData(string appSecret) : this(appSecret, 
-            new MainApiConnection(Settings.GetMainApiConnectionSettings()),
-            new KeysApiConnection(Settings.GetKeysApiConnectionSettings()))
-        {
-            _keysApiConnection.HttpWebRequestFactory = new UnityWebRequestFactory();
-        }
-
-        public RemoteMetaData(string appSecret, MainApiConnection mainApiConnection, KeysApiConnection keysApiConnection)
+        public RemoteMetaData(string appSecret)
         {
             Checks.ArgumentNotNullOrEmpty(appSecret, "appSecret");
-            Checks.ArgumentNotNull(mainApiConnection, "mainApiConnection");
-            Checks.ArgumentNotNull(keysApiConnection, "keysApiConnection");
 
             DebugLogger.LogConstructor();
             DebugLogger.LogVariable(appSecret, "appSecret");
 
             _appSecret = appSecret;
-            _mainApiConnection = mainApiConnection;
-            _keysApiConnection = keysApiConnection;
+            _mainApiConnection = new MainApiConnection(Settings.GetMainApiConnectionSettings());
+
+            var keysSettings = Settings.GetKeysApiConnectionSettings();
+
+            string overrideKeysUrl;
+
+            if (TryReadEnvironmentVariable("PK_PATCHER_KEYS_URL", out overrideKeysUrl))
+            {
+                bool useHttps = overrideKeysUrl.StartsWith("https://");
 
-            _keysApiConnection.HttpWebRequestFactory = new UnityWebRequestFactory();
+                overrideKeysUrl = overrideKeysUrl.Replace("https://", string.Empty)
```

Comment:
That's quite incorrect method of getting the host name. URL can contain host & port. You should use Uri class https://msdn.microsoft.com/en-us/library/system.uri(v=vs.110).aspx to extract host and port. Also Sheme field tells you if this is http or https.